### PR TITLE
Add script to auto-generate chain id -> rpc endpoints mapping files

### PR DESCRIPTION
**Pull request by derekpierre**
- Add script to auto-generate json endpoint mapping files for any domain (`lynx`, `tapir`, `mainnet`)
   - The script runs a health check on each rpc endpoint checking chain id and drift of current block time
   - The script runs using asyncio to speed up processing
- Expand the chain ids supported by `lynx`
- Added pre-commit hooks and linting

### CODE REVIEW

File: scripts/generate_chain_endpoints_mapping.py [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+    84532,  # Base Sepolia Testnet
+    421614,  # Arbitrum Sepolia
+    534351,  # Scroll Sepolia Testnet
+    534352,  # Scroll
+    3141592,  # Filecoin - Butterfly testnet
+    11155111,  # Sepolia
+    11155420,  # OP Sepolia Testnet
+}
+
+EXTRA_KNOWN_RPC_ENDPOINTS = {
```
Comment (theref) [inline scripts/generate_chain_endpoints_mapping.py]: love that you'd added drpc endpoints here. Similary I think we should add some from https://www.nodies.app/ which is built on top of POKT (similar to dRPC) - although, as I type this, I can't remember if pokt endpoints are compatible with TACo...

`https://ethereum-sepolia-public.nodies.app`

**Review by theref — COMMENTED** [review]
Given that we're starting to add some proper scripts in here, how do you feel about a `pre-commit` file that runs black/ruff/etc and also runs the `sort_chain_endpoints_files.py`

File: scripts/generate_chain_endpoints_mapping.py (hunk 1 above)
Comment (derekpierre) [reply scripts/generate_chain_endpoints_mapping.py]: ✔️

File: .github/workflows/darker.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+name: 🧹 Darker
```
Comment (theref) [inline .github/workflows/darker.yml]: We can just use black - we use darker on `nucypher` so that we don't end up with one massive formatting commit

Comment (derekpierre) [reply .github/workflows/darker.yml]: ✔️

**Review by theref — APPROVED** [review]

File: scripts/generate_chain_endpoints_mapping.py (hunk 1 above)
Comment (derekpierre) [reply scripts/generate_chain_endpoints_mapping.py]: > I can't remember if pokt endpoints are compatible with TACo...

There was an issue before, but it was fixed via https://github.com/nucypher/nucypher/pull/3491 for nodes to be agnostic to RPC endpoints.

**Pull request merged**